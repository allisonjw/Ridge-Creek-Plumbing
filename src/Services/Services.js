import React from 'react';
import './Services.scss';
import servicesImage from '../images/plumbing-equipment.jpg';

export const Services = () => {
    return (
      <section className="section_services">
        <div className="services_banner" 
        aria-label="notification of services page">
          <div className="services_img" aria-label="" style={{ backgroundImage:`linear-gradient(rgba(255, 252, 252, 0.25), rgba(255, 254, 254, 0.15)), url(${servicesImage})` }}>
            <h2 className="portfolio_banner-h2">
              Services
            </h2>
          </div>
        </div>      
      </section>
    )
};

export default Services;