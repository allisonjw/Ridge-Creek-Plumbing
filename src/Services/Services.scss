@import '../SCSS/variables';
@import '../SCSS/media-queries';

.section_services {
    // animation: fade-in 1s ease-out both 0.3s;
    // -webkit-animation: fade-in 1s ease-out both 0.3s;
    // -moz-animation: fade-in 1s ease-out both 0.3s;
    // -o-animation: fade-in 1s ease-out both 0.3s;
}

.services_banner {
    width: 90%;
    height: 70vh;
    margin: 2em auto;
    position: relative;
    border-bottom: 1px solid #000;
    padding-bottom: 1%;
}

.services_img {
    width: 100%;
    height: 70vh;
    background-size: cover;
    background-attachment: fixed;
    background-position: 50% 50%;
    background-repeat: no-repeat;
    .services_banner-h2 {
        position: absolute;
        font-size: 4em;
        color: white;
        top: 18%;
        left: 14%;
        text-shadow: 5px 4px 6px #000000b5;
    }
}

.services_description-paragraph {
    color: $main_color;
    font-size: 1.4em;
    line-height: 1.7;
    text-align: center;
    font-weight: 500;
    margin: 0 5%;
    padding-bottom: 2%;
    .services_union-link {
        margin-left: 5px;
        text-decoration: none;
        font-weight: 700;
        color: $accent_color;
        &:hover {
            cursor: pointer;
            color: #0a0a94f2;        
        }
    }
}

.services_body {
    margin-right: auto;
    margin-left: auto;
    padding-left: 15px;
    padding-right: 15px;
}

.services_article {
    margin: 0 auto 5% auto;
    padding-left: 15px;
    padding-right: 15px;
    display: flex;
}

.services_img-holder-1,
.services_img-holder-2,
.services_img-holder-3,
.services_img-holder-4 {
    float: left;
    // animation: fadeIn 1s ease-out both 0.2s;
}

.services_image-1,
.services_image-2,
.services_image-3,
.services_image-4 {
    max-width: 50vw;
    box-shadow: 9px 8px 12px 0px #00000085;
    border-radius: 3px;
}

.services_img-holder-1 {
    float: left;
    .services_image-1 {
        float: right;
    }
}

.services_h3 {
    font-size: 1.6em;
    margin-bottom: 1%;
    text-transform: uppercase;
    color: $accent_color;
    letter-spacing: 2px;
}

.services_description-holder-1,
.services_description-holder-2,
.services_description-holder-3,
.services_description-holder-4 {
    padding: 5% 0 0 5%;
    // animation: fadeIn 1s ease-out both 0.2s;
}

.services_paragraph {
    font-size: 1.4em;
    line-height: 1.4;
    font-weight: 500;
    color: $main_color;
}

@keyframes fadeIn {
    from {opacity: 0%;}
    to {opacity: 100%;}
}


