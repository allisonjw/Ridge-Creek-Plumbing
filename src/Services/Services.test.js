import React from 'react';
import Services from './Services';
import { shallow } from 'enzyme';

describe('Services', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Services />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});