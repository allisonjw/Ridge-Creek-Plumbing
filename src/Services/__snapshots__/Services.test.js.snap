// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Services should match the snapshot 1`] = `
<section
  className="section_services"
>
  <Fade
    left={true}
  >
    <div
      aria-label="notification of services page"
      className="services_banner"
    >
      <div
        aria-label=""
        className="services_img"
        style={
          Object {
            "backgroundImage": "linear-gradient(rgba(255, 252, 252, 0.25), rgba(255, 254, 254, 0.15)), url(plumbing-equipment.jpg)",
          }
        }
      >
        <h2
          className="services_banner-h2"
        >
          Our Services
        </h2>
      </div>
    </div>
  </Fade>
  <p
    className="services_description-paragraph"
  >
    For light commercial projects, tenant improvement projects and larger commercial and industrial projects, as a signatory to the Local 26 Plumbers and Pipefitters Union, we have access to a pool of trained and skilled plumbers who can assist with any plumbing project. Visit Local 26's
    <a
      aria-label="link to Local Union 26's website"
      className="services_union-link"
      href="http://www.ua26.org/"
      rel="noopener noreferrer"
      target="_blank"
    >
      website
    </a>
    <br />
    <br />
    Local 26 is part of a National Plumbers and Pipefitters Union. Additional information about the National UA can be found
    <a
      aria-label="link to National Plumber's website"
      className="services_union-link"
      href="http://www.ua.org/"
      rel="noopener noreferrer"
      target="_blank"
    >
      here
    </a>
  </p>
  <section
    className="services_body"
  >
    <article
      className="services_article"
    >
      <Reveal
        count={1}
        countOut={1}
        delay={0}
        delayOut={0}
        duration={1000}
        durationOut={1000}
        effect="pulse"
        inEffect={
          Object {
            "count": 1,
            "delay": 0,
            "duration": 1000,
            "forever": undefined,
            "make": [Function],
            "reverse": undefined,
            "style": Object {
              "animationFillMode": "both",
            },
          }
        }
        outEffect={
          Object {
            "count": 1,
            "delay": 0,
            "duration": 1000,
            "forever": undefined,
            "make": [Function],
            "reverse": undefined,
            "style": Object {
              "animationFillMode": "both",
            },
          }
        }
      >
        <aside
          className="services_img-holder-1"
        >
          <img
            alt="plumbing tools"
            className="services_image-1"
            src="plumbing-585658_640.png"
          />
        </aside>
      </Reveal>
      <aside
        className="services_description-holder-1"
      >
        <h3
          className="services_h3"
        >
          Plumbing
        </h3>
        <p
          className="services_paragraph"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
        </p>
      </aside>
    </article>
    <article
      className="services_article"
    >
      <aside
        className="services_description-holder-2"
      >
        <h3
          className="services_h3"
        >
          Remodeling
        </h3>
        <p
          className="services_paragraph"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
        </p>
      </aside>
      <Reveal
        count={1}
        countOut={1}
        delay={0}
        delayOut={0}
        duration={1000}
        durationOut={1000}
        effect="pulse"
        inEffect={
          Object {
            "count": 1,
            "delay": 0,
            "duration": 1000,
            "forever": undefined,
            "make": [Function],
            "reverse": undefined,
            "style": Object {
              "animationFillMode": "both",
            },
          }
        }
        outEffect={
          Object {
            "count": 1,
            "delay": 0,
            "duration": 1000,
            "forever": undefined,
            "make": [Function],
            "reverse": undefined,
            "style": Object {
              "animationFillMode": "both",
            },
          }
        }
      >
        <aside
          className="services_img-holder-2"
        >
          <img
            alt="tools for construction work"
            className="services_image-2"
            src="construction.png"
          />
        </aside>
      </Reveal>
    </article>
    <article
      className="services_article"
    >
      <Reveal
        count={1}
        countOut={1}
        delay={0}
        delayOut={0}
        duration={1000}
        durationOut={1000}
        effect="pulse"
        inEffect={
          Object {
            "count": 1,
            "delay": 0,
            "duration": 1000,
            "forever": undefined,
            "make": [Function],
            "reverse": undefined,
            "style": Object {
              "animationFillMode": "both",
            },
          }
        }
        outEffect={
          Object {
            "count": 1,
            "delay": 0,
            "duration": 1000,
            "forever": undefined,
            "make": [Function],
            "reverse": undefined,
            "style": Object {
              "animationFillMode": "both",
            },
          }
        }
      >
        <aside
          className="services_img-holder-3"
        >
          <img
            alt="copper piping for plumbing"
            className="services_image-3"
            src="copper-pipe.png"
          />
        </aside>
      </Reveal>
      <aside
        className="services_description-holder-3"
      >
        <h3
          className="services_h3"
        >
          Re-piping
        </h3>
        <p
          className="services_paragraph"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
        </p>
      </aside>
    </article>
    <article
      className="services_article"
    >
      <aside
        className="services_description-holder-4"
      >
        <h3
          className="services_h3"
        >
          Water and Gas
        </h3>
        <p
          className="services_paragraph"
        >
          Whether you have a new water heater or an older water heater, proper and routine water heater maintenance is something you can do to extend its life, minimize costs for repair and replacement, and lower your energy costs.
        </p>
      </aside>
      <Reveal
        count={1}
        countOut={1}
        delay={0}
        delayOut={0}
        duration={1000}
        durationOut={1000}
        effect="pulse"
        inEffect={
          Object {
            "count": 1,
            "delay": 0,
            "duration": 1000,
            "forever": undefined,
            "make": [Function],
            "reverse": undefined,
            "style": Object {
              "animationFillMode": "both",
            },
          }
        }
        outEffect={
          Object {
            "count": 1,
            "delay": 0,
            "duration": 1000,
            "forever": undefined,
            "make": [Function],
            "reverse": undefined,
            "style": Object {
              "animationFillMode": "both",
            },
          }
        }
      >
        <aside
          className="services_img-holder-4"
        >
          <img
            alt="gas flame from a stove"
            className="services_image-4"
            src="appliance-gas.png"
          />
        </aside>
      </Reveal>
    </article>
  </section>
</section>
`;
